### PR TITLE
Created the Clustering module

### PR DESCRIPTION
**Pull request by b-sherman**
+Added clustering rst
+Added clustering images
+Added clustering code

### CODE REVIEW

**Comment by BroccoliHijinx** [comment]
1. In the code, it might be more effective to motivate it with a real example, like the toy manufacturer in the write-up.  Data could be something like age, toys bought in the last month.  Having the data form a real example can probably help the average reader wrap their head around it.  Also (and I am not sure if it is possible/easy), if an approximate class divider can be added to the graph it would help the readers.

2. Since we don't have an introduction section to unsupervised learning (that I know of), it might be useful to define the term in your overview (since this section is first alphabetically).  

3. For K-means, can you expand on the point of degenerate starting points?  As in, how are initial cluster centers chosen, and is there a good way to make sure you get the best ones?

4. For hierarchical clustering, can you explain how clusters are chosen in top-down and bottom up clustering?  Would an algorithm consider every split of the largest (or every combination of smallest) clusters?  Also, can you explain why sklearn calls it agglomerative? What is different about that word from hierarchical?

**Comment by astorfi** [comment]
@b-sherman Thank you for creating this

**Comment by astorfi** [comment]
@weric16 Thank you for your review

**Comment by b-sherman** [comment]
Updated the module with the review suggestions.

File: docs/source/content/unsupervised/_img/clustering.rst [hunk 1]
```diff
@@ -0,0 +1,222 @@
+##########
```
Comment (mergu) [inline docs/source/content/unsupervised/_img/clustering.rst]: you accidentally copied this to the img folder it seems

Comment (b-sherman) [reply docs/source/content/unsupervised/_img/clustering.rst]: Yep. Thanks for catching that. I just removed the extra copy.

**Comment by astorfi** [comment]
@mergu @weric16 @hokie45 Can someone review and approve this pull request?

**Review by mergu — APPROVED** [review]
Looks great!

File: docs/source/content/unsupervised/clustering.rst [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+display the final plot shown in *Figure 9*.
+
+.. figure:: _img/Hierarchical.png
+
+   **Figure 9. A final clustered data set**
+
+The clusters are color coded and large clusters are surrounded with a border
+to show which data points belong to them.
+
+Summary
```
Comment (hokie45) [inline docs/source/content/unsupervised/clustering.rst:211]: The summary title should have '====' above also to keep it out of the Methods section

**Review by hokie45 — CHANGES_REQUESTED** [review]

**Review by hokie45 — APPROVED** [review]

**Pull request merged**